src/sidebar/helpers/strip-internal-properties.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Return a shallow clone of `obj` with all client-only properties removed.
+ * Client-only properties are marked by a '$' prefix.
+ */
+export function stripInternalProperties(
+  obj: Record<string, unknown>
+): Record<string, unknown> {
+  const result: Record<string, unknown> = {};
+  for (const [key, value] of Object.entries(obj)) {
+    if (!key.startsWith('$')) {
+      result[key] = value;
+    }
+  }
+  return result;
```

Comment:
I wonder if the typecasting would make more sense on this `return`, e.g. `return result as Annotation`;